gapic/src/main/com/google/gapid/perfetto/models/CounterTrack.java

```diff
@@ -39,17 +40,21 @@
 
 public class CounterTrack extends Track.WithQueryEngine<CounterTrack.Data> {
   private static final String VIEW_SQL_DELTA =
-      "select ts + 1 ts, lead(ts) over win - ts dur, lead(value) over win value " +
+      "select ts + 1 ts, lead(ts) over win - ts dur, lead(value) over win value, id " +
       "from counter where track_id = %d window win as (order by ts)";
   private static final String VIEW_SQL_EVENT =
-      "select ts, lead(ts, 1, (select end_ts from trace_bounds)) over win - ts dur, value " +
+      "select ts, lead(ts, 1, (select end_ts from trace_bounds)) over win - ts dur, value, id " +
       "from counter where track_id = %d window win as (order by ts)";
   private static final String SUMMARY_SQL =
-      "select min(ts), max(ts + dur), avg(value) from %s group by quantum_ts";
-  private static final String COUNTER_SQL = "select ts, ts + dur, value from %s";
-  private static final String VALUE_SQL = "select ts, ts + dur, value from %s where ts = %d";
+      "select min(ts), max(ts + dur), avg(value), best_id from " +
+        "(select *, first_value(id) over (partition by quantum_ts order by dur desc) as best_id from %s) " +
+      "group by quantum_ts";
+  private static final String COUNTER_SQL = "select ts, ts + dur, value, id from %s";
+  private static final String VALUE_SQL =
+      "select ts, ts + dur, value, id from %s " +
+      "where ts <= %d and ts + dur > %d";
   private static final String RANGE_SQL =
-      "select ts, ts + dur, value from %s " +
+      "select ts, ts + dur, value, id from %s " +
```

Comment:
Is the ts + dur under select getting used? If not, we can remove that from the query.

Reply:
It's actually used at `data.ts[rows] = res.getLong(rows - 1, 1, 0);`.